apps/dashboard/src/app/nebula-app/(app)/components/ChatPageContent.tsx

```diff
@@ -85,17 +85,33 @@ export function ChatPageContent(props: {
     NebulaContext | undefined
   >(() => {
     const contextRes = props.session?.context;
-    const value: NebulaContext = {
-      chainIds: contextRes?.chain_ids || null,
-      walletAddress: contextRes?.wallet_address || null,
+    
+    // If we have context from an existing session, use that
+    if (contextRes) {
+      return {
+        chainIds: contextRes.chain_ids || null,
+        walletAddress: contextRes.wallet_address || null,
+      };
+    }
+    
+    // For new sessions, try to get cached chains from localStorage
+    const cachedChains = localStorage.getItem('nebula-cached-chains');
+    const parsedCachedChains = cachedChains ? JSON.parse(cachedChains) : null;
+    
+    return {
+      chainIds: parsedCachedChains || ['1'], // Default to Ethereum if no cache
```

Comment:
Don't need to set a default chain here - default is always whatever the user's wallet is connected to